docs/guides/composer.mdx

```diff
@@ -0,0 +1,731 @@
+<td><code>OPENQASM 2.0;</code></td>
+</tr>
+<tr>
+<td><code>qreg name[size];</code></td>
+<td>Declare a named register of qubits</td>
+<td><code>qreg q[5];</code></td>
+</tr>
+<tr>
+<td><code>creg name[size];</code></td>
+<td>Declare a named register of bits</td>
+<td><code>creg c[5];</code></td>
+</tr>
+<tr>
+<td><code>include “filename”;</code></td>
+<td>Open and parse another source file</td>
+<td><code>include “qelib1.inc”;</code></td>
+</tr>
+<tr>
+<td><code>gate name(params) qargs</code></td>
+<td>Declare a unitary gate</td>
+<td>(see text of paper)</td>
+</tr>
+<tr>
+<td><code>opaque name(params) qargs;</code></td>
+<td>Declare an opaque gate</td>
+<td>(see text of paper)</td>
+</tr>
+<tr>
+<td><code>// comment text</code></td>
+<td>Comment a line of text</td>
+<td><code>// oops!</code></td>
+</tr>
+<tr>
+<td><code>U(theta,phi,lambda) qubit|qreg;</code></td>
+<td>Apply built-in single-qubit gate(s) (see [b])</td>
+<td><code>U(pi/2,2\*pi/3,0) q[0];</code></td>
+</tr>
+<tr>
+<td><code>CX qubit|qreg,qubit|qreg;</code></td>
+<td>Apply built-in CNOT gate(s)</td>
+<td><code>CX q[0],q[1];</code></td>
+</tr>
+<tr>
+<td><code>measure qubit|qreg -&gt; bit|creg;</code></td>
+<td>Make measurement(s) in $Z$ basis</td>
+<td><code>measure q -&gt; c;</code></td>
+</tr>
+<tr>
+<td><code>reset qubit|qreg;</code></td>
+<td>Prepare qubit(s) in $\vert 0\rangle$</td>
+<td><code>reset q[0];</code></td>
+</tr>
+<tr>
+<td><code>gatename(params) qargs;</code></td>
+<td>Apply a user-defined unitary gate</td>
+<td><code>crz(pi/2) q[1],q[0];</code></td>
+</tr>
+<tr>
+<td><code>if(creg==int) qop;</code></td>
+<td>Conditionally apply quantum operation</td>
+<td><code>if(c==5) CX q[0],q[1];</code></td>
+</tr>
+<tr>
+<td><code>barrier qargs;</code></td>
+<td>Prevent transformations across this source line</td>
+<td><code>barrier q[0],q[1];</code></td>
+</tr>
+</tbody></table>
+
+
+\[a] This must appear as the first non-comment line of the file.
+
+\[b] The parameters theta, phi, and lambda are given by *parameter expressions*; for more information, see [page 5 of the paper](https://arxiv.org/abs/1707.03429#page=5) and [Appendix A](https://arxiv.org/pdf/1707.03429.pdf#page=21).
+
+#### Create a custom operation in OpenQASM
+
+You can define new unitary operations in the code editor (see the figure below for an example). The operations are applied using the statement `name(params) qargs;` just like the built-in operations. The parentheses are optional if there are no parameters.
+
+To define a custom operation, enter it in the OpenQASM code editor using this format: `gatename(params) qargs;`. If you click **+Add** in the operations list, you will be prompted to enter a name for your custom operation, which you can then build in the code editor.
+
+Once you have defined your custom operation, drag it onto the graphical editor and use the edit icon to fine-tune the parameters.
+
+a) The gates to be included in the custom operation:
```

Comment:
I was going to suggest using a dedicated list, but I realize the images will make that tricky. Would it make more sense to use captions instead on the images? https://github.com/Qiskit/documentation/blob/main/mdx-guide.md#images

Reply:
<img width="767" alt="Screenshot 2025-06-23 at 7 46 11 PM" src="https://github.com/user-attachments/assets/00652031-5b01-4a38-b0a0-3f058c26b673" />
 
Does this seem better?

Reply:
Yes on desktop view! Maybe not as nice for mobile / small screens because there could be horizontal scroll. Why not use captions?